reflex/components/component.py

```diff
@@ -327,10 +328,22 @@ def __init__(self, *args, **kwargs):
                     # If it is not a valid var, check the base types.
                     passed_type = type(value)
                     expected_type = fields[key].outer_type_
-                if not types._issubclass(passed_type, expected_type):
+                if types.is_union(passed_type):
```

Comment:
Can we add some comments on what these checks are doing at a high level / how the passed_types work

Reply:
Of course, done